### PR TITLE
fix(Commerce SSR): export the appropriate sort criteria for commerce

### PR DESCRIPTION
**Pull request by fpbrault**
https://coveord.atlassian.net/browse/KIT-3700

### CODE REVIEW

File: packages/headless/src/ssr-commerce.index.ts [hunk 1]
```diff
@@ -362,6 +362,9 @@ export {
   buildQueryRankingExpressionSortCriterion,
   buildRelevanceSortCriterion,
 } from './features/sort-criteria/criteria.js';
+export type {SortCriterion as SortSortCriterion} from './features/sort/sort.js';
+
+export {SortBy as SortSortBy} from './features/sort/sort.js';
```
Comment (fpbrault) [inline packages/headless/src/ssr-commerce.index.ts:367]: I'm not sure what could be better names to distinguish these from the sort-criteria.js exports

**Review by alexprudhomme — COMMENTED** [review]
I am pretty sure you should just need to export from `features/commerce/sort/... `instead of `features/sort-criteria/...`. Wouldn't that work ? 
```
export type {
  SortCriterion,
  SortBy} from './features/commerce/sort/sort.ts'

```

File: packages/headless/src/ssr-commerce.index.ts (hunk 1 above)
Comment (fpbrault) [reply packages/headless/src/ssr-commerce.index.ts:367]: I was ending up with a hydration error, but I just figured out why

File: packages/samples/headless-ssr-commerce/components/sort.tsx [hunk 2]
```diff
@@ -10,15 +11,19 @@ export default function Sort() {
...
+    switch (criterion.by) {
+      case SortBy.Relevance:
+        return 'Relevance';
+      case SortBy.Fields:
+        return criterion.fields.map((field) =>
+          field
+            ? (field.displayName ??
+              [field.name, field.direction].filter(Boolean).join(' '))
+            : ''
+        );
```
Comment (louis-bompart) [inline packages/samples/headless-ssr-commerce/components/sort.tsx:24]: Should probably refactor the map callback into a separate function for readability

**Review by louis-bompart — APPROVED** [review]
nit

**Review by alexprudhomme — APPROVED** [review]

**Pull request merged**